identity/handler_test.go

```diff
@@ -991,4 +992,108 @@ func TestHandler(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("case=should delete credential of a specific user and no longer be able to retrieve it", func(t *testing.T) {
+		createIdentities := func(identities map[identity.CredentialsType]string) func(t *testing.T) *identity.Identity {
+			return func(t *testing.T) *identity.Identity {
+				i := identity.NewIdentity("")
+				for ct, config := range identities {
+					i.SetCredentials(ct, identity.Credentials{
+						Type:   ct,
+						Config: sqlxx.JSONRawMessage(config),
+					})
+				}
+				i.Traits = identity.Traits("{}")
+				require.NoError(t, reg.Persister().CreateIdentity(context.Background(), i))
+				return i
+			}
+		}
+
+		for name, ts := range map[string]*httptest.Server{"public": publicTS, "admin": adminTS} {
+			t.Run("type=remove unknown identity/"+name, func(t *testing.T) {
+				remove(t, ts, "/identities/"+x.NewUUID().String()+"/credentials/azerty", http.StatusNotFound)
+			})
+			t.Run("type=remove unknown type/"+name, func(t *testing.T) {
+				i := createIdentities(map[identity.CredentialsType]string{identity.CredentialsTypePassword: `{"secret":"pst"}`})(t)
+				remove(t, ts, "/identities/"+i.ID.String()+"/credentials/azerty", http.StatusNotFound)
+			})
+			t.Run("type=remove password type/"+name, func(t *testing.T) {
+				i := createIdentities(map[identity.CredentialsType]string{identity.CredentialsTypePassword: `{"secret":"pst"}`})(t)
+				remove(t, ts, "/identities/"+i.ID.String()+"/credentials/password", http.StatusBadRequest)
+			})
+			t.Run("type=remove oidc type/"+name, func(t *testing.T) {
+				i := createIdentities(map[identity.CredentialsType]string{identity.CredentialsTypeOIDC: `{"id":"pst"}`})(t)
+				remove(t, ts, "/identities/"+i.ID.String()+"/credentials/oidc", http.StatusBadRequest)
+			})
+			t.Run("type=remove webauthn passwordless type/"+name, func(t *testing.T) {
+				expected := `{"credentials":[{"id":"THTndqZP5Mjvae1BFvJMaMfEMm7O7HE1ju+7PBaYA7Y=","added_at":"2022-12-16T14:11:55Z","public_key":"pQECAyYgASFYIMJLQhJxQRzhnKPTcPCUODOmxYDYo2obrm9bhp5lvSZ3IlggXjhZvJaPUqF9PXqZqTdWYPR7R+b2n/Wi+IxKKXsS4rU=","display_name":"test","authenticator":{"aaguid":"rc4AAjW8xgpkiwsl8fBVAw==","sign_count":0,"clone_warning":false},"is_passwordless":true,"attestation_type":"none"}],"user_handle":"Ef5JiMpMRwuzauWs/9J0gQ=="}`
+				i := createIdentities(map[identity.CredentialsType]string{identity.CredentialsTypeWebAuthn: expected})(t)
+				remove(t, ts, "/identities/"+i.ID.String()+"/credentials/webauthn", http.StatusNoContent)
+				// Check that webauthn has not been deleted
+				res := get(t, ts, "/identities/"+i.ID.String(), http.StatusOK)
+				assert.EqualValues(t, i.ID.String(), res.Get("id").String(), "%s", res.Raw)
+				assert.True(t, res.Get("credentials").Exists())
+				assert.True(t, res.Get("credentials").Get("webauthn").Exists())
```

Comment:
Could you please refactor this to use `GetIdentityConfidential` instead, and then ensure that the credentials config is correct - it should have the same value as `expected`. That ensures this works properly

Reply:
Done, I used the snapshot method (rather than comparing to expected) as it seams that's what's done on other tests